qrexec/tests/socket/agent.py

```diff
@@ -438,6 +438,41 @@ def test_exec_service_with_arg(self):
             ],
         )
 
+    def test_connect_socket_no_metadata(self):
+        socket_path = os.path.join(
+            self.tempdir, "rpc", "qubes.SocketService+arg2"
+        )
+        with open(
+            os.path.join(self.tempdir, "rpc-config", "qubes.SocketService+arg2"), "w"
+        ) as f:
+            f.write("""\
+skip-service-descriptor = true
+""")
+        server = qrexec.socket_server(socket_path)
+        self.addCleanup(server.close)
+
+        target = self.execute_qubesrpc("qubes.SocketService+arg2", "domX")
+
+        server.accept()
+
+        message = b"stdin data"
+        target.send_message(qrexec.MSG_DATA_STDIN, message)
+        target.send_message(qrexec.MSG_DATA_STDIN, b"")
+        self.assertEqual(server.recvall(len(message) - 1), message[:-1])
```

Comment:
Why -1?

Reply:
I was trying to figure out why the test was hanging.  It turns out that it was failing to load the config file.